### PR TITLE
Create a JPanel that can trigger UI DSL validation

### PR DESCRIPTION
**Pull request by abrooksv**
Check the usage in the tests, happy to tweak it

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                updateErrorInfo(performValidation())
+                initValidation()
+            }
+        }
+        validationAlarm.addRequest(validateRequest, VALIDATION_INTERVAL, ModalityState.stateForComponent(this))
+    }
+
+    open inner class ButtonAction internal constructor(name: String, private val listener: (event: ActionEvent) -> Unit) : AbstractAction(name) {
+        override fun actionPerformed(e: ActionEvent) {
+            if (performingAction) return
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt:121]: I'm not sure this is actually worth doing tbh...any asyc suspend that is kicked off won't actually block the button

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+                }
+
+                validator.updateInfo(it)
+            }
+        }
+    }
+
+    private fun isDisposed(): Boolean = Disposer.isDisposed(disposable)
+
+    private fun runOnUi(action: () -> Unit) {
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt:182]: Not that I could find

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+fun validatingPanel(disposable: Disposable, init: ValidatingPanelBuilder.() -> Unit): ValidatingPanel {
+    val ref = Ref<ValidatingPanelBuilderImpl>()
+    val contentPanel = panel {
+        val builder = ValidatingPanelBuilderImpl(this)
+        builder.init()
+
+        ref.set(builder)
+    }
+
+    return ref.get().build(disposable, contentPanel)
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt:200]: why do we need the ref?

**Review by rli — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt (hunk 3 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt]: Removed it.

**Review by rli — APPROVED** [review]

**Pull request merged**